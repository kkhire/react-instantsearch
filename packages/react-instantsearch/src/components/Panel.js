import React from 'react';
import PropTypes from 'prop-types';

const Panel = ({ children, canRefine, cx, renderHeader, renderFooter }) => (
  <div className={cx('', !canRefine && '-noRefinement').trim()}>
    {renderHeader && (
      <div className={`${cx('header')} ais-header`.trim()}>
        {renderHeader()}
      </div>
    )}

    <div className={`${cx('body')} ais-body`.trim()}>{children}</div>

    {renderFooter && (
      <div className={`${cx('footer')} ais-footer`.trim()}>
        {renderFooter()}
      </div>
    )}
  </div>
);

Panel.propTypes = {
  children: PropTypes.node.isRequired,
  cx: PropTypes.func,
  canRefine: PropTypes.bool,
  renderHeader: PropTypes.func,
  renderFooter: PropTypes.func,
};

Panel.defaultProps = {
  cx: () => '',
  canRefine: true,
};

export default Panel;
