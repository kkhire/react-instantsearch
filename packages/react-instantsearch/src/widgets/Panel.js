import Panel from '../components/Panel';

/**
 * The Panel widget wraps other widgets in a consistent panel design. It's shipped by default on all the widget, but we still expose it as a standalone widget to enable to possibility to wrap your custom widget with it.
 *
 * It also reacts, indicates and set CSS classes when widgets are no more
 * relevant for refining. E.g. when a RefinementList becomes empty because of
 * the current search results.
 * @name Panel
 * @kind widget
 * @propType {boolean} [canRefine=true] - Indicate if the widget can be refine
 * @propType {function} [renderHeader] - Adds a header to the Panel.
 * @propType {function} [renderFooter] - Adds a footer to the Panel.
 * @themeKey ais-Panel - the root div of the Panel
 * @themeKey ais-Panel--noRefinement - the root div of the Panel without refinement
 * @themeKey ais-Panel-header - the header of the Panel (optional)
 * @themeKey ais-Panel-body - the body of the Panel
 * @themeKey ais-Panel-footer - the footer of the Panel (optional)
 * @example
 * import React from 'react';
 * import { InstantSearch, Panel } from 'react-instantsearch/dom';
 * import { connectRefinementList } from 'react-instantsearch/connectors';
 *
 * const MyCustomRefinementList = connectRefinementList(({ renderHeader }) => (
 *   <Panel renderHeader={renderHeader}>
 *     My custom refinement list
 *   </Panel>
 * ));
 *
 * export default function App() {
 *   return (
 *     <InstantSearch
 *       appId="latency"
 *       apiKey="6be0576ff61c053d5f9a3225e2a90f76"
 *       indexName="ikea"
 *     >
 *       <MyCustomRefinementList
 *         attributeName="category"
 *         renderHeader={() => 'Category'}
 *       />
 *     </InstantSearch>
 *   );
 * }
 */
export default Panel;
