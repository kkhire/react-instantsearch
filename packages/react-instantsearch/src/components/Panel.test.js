import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Panel from './Panel';

Enzyme.configure({ adapter: new Adapter() });

describe('Panel', () => {
  const defaultProps = {
    cx: (...args) => args.filter(x => Boolean(x)).join(' '),
  };

  it('expect to render', () => {
    const wrapper = shallow(
      <Panel>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with custom cx', () => {
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(
      <Panel {...props}>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render without refinement', () => {
    const props = {
      ...defaultProps,
      canRefine: false,
    };

    const wrapper = shallow(
      <Panel {...props}>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with header', () => {
    const props = {
      ...defaultProps,
      renderHeader: () => 'Header',
    };

    const wrapper = shallow(
      <Panel {...props}>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with footer', () => {
    const props = {
      ...defaultProps,
      renderFooter: () => 'Footer',
    };

    const wrapper = shallow(
      <Panel {...props}>
        <div>Hello content</div>
      </Panel>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
