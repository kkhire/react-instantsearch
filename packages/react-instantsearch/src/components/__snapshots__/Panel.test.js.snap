// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel expect to render 1`] = `
<div
  className=""
>
  <div
    className="ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render with custom cx 1`] = `
<div
  className=""
>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render with footer 1`] = `
<div
  className=""
>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
  <div
    className="footer ais-footer"
  >
    Footer
  </div>
</div>
`;

exports[`Panel expect to render with header 1`] = `
<div
  className=""
>
  <div
    className="header ais-header"
  >
    Header
  </div>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`Panel expect to render without refinement 1`] = `
<div
  className="-noRefinement"
>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;
